src/test/java/picard/sam/markduplicates/UmiAwareMarkDuplicatesWithMateCigarTest.java

```diff
@@ -322,14 +331,84 @@ public void testUmiMetrics(List<String> umis, List<String> assignedUmi, final Li
+                      2,                         // DUPLICATE_SETS_WITHOUT_UMI
+                      4,                         // DUPLICATE_SETS_WITH_UMI
+                      effectiveLength2_1,        // INFERRED_UMI_ENTROPY
+                      effectiveLength1_1_1,      // OBSERVED_UMI_ENTROPY
+                      estimatedBaseQuality1_12,  // ESTIMATED_BASE_QUALITY_OF_UMIS
+                      0),                        // UMI_WITH_N
+
+                   new UmiMetrics("B",           // LIBRARY
+                      4.0,                       // MEAN_UMI_LENGTH
+                      1,                         // OBSERVED_UNIQUE_UMIS
+                      1,                         // INFERRED_UNIQUE_UMIS
+                      0,                         // OBSERVED_BASE_ERRORS
+                      2,                         // DUPLICATE_SETS_WITHOUT_UMI
+                      2,                         // DUPLICATE_SETS_WITH_UMI
+                      0.0,                       // INFERRED_UMI_ENTROPY
+                      0.0,                       // OBSERVED_UMI_ENTROPY
+                      -1,                        // ESTIMATED_BASE_QUALITY_OF_UMIS
+                      0)                         // UMI_WITH_N
+                )
+        }};
+    }
+
+    @Test(dataProvider = "testMultipleLibraryUmiMetricsDataProvider")
+    public void testMultipleLibraryUmiMetrics(final List<String> libraries, final List<String> umis, final List<String> assignedUmi, final List<Boolean> isDuplicate,
+                               final int editDistanceToJoin, final List<UmiMetrics> expectedMetricsList) {
+
+        // Test collection of UMI metrics across multiple libraries
+        final Map<String, List<String>> expectedAssignedUmis = new HashMap<>();
+        for (int i = 0;i < umis.size();i++) {
+            // Get assigned UMIs for each particular library
+            expectedAssignedUmis.putIfAbsent(libraries.get(i), new ArrayList<>());
+            expectedAssignedUmis.get(libraries.get(i)).add(assignedUmi.get(i));
+        }
+
+        // Evaluate UMI metrics over each library
+        for (final UmiMetrics expectedMetrics : expectedMetricsList) {
+            final UmiAwareMarkDuplicatesWithMateCigarTester tester = getTester(false);
+            tester.addArg("MAX_EDIT_DISTANCE_TO_JOIN=" + editDistanceToJoin);
+            for (int i = 0;i < umis.size();i++) {
+                if(expectedMetrics.LIBRARY.equals(libraries.get(i))) {
+                    tester.addMatePairWithUmi(libraries.get(i), umis.get(i), assignedUmi.get(i), isDuplicate.get(i), isDuplicate.get(i));
+                }
+            }
+
+            tester.setExpectedAssignedUmis(expectedAssignedUmis.get(expectedMetrics.LIBRARY));
+            tester.setExpectedMetrics(expectedMetrics);
+            tester.runTest();
+        }
+    }
+
     @DataProvider(name = "testUmiUtilDataProvider")
     private Object[][] testUmiUtilDataProvider() {
```

Comment:
I know this isn't part of the changed code, but it looks this test uses data provider but doesn't use the data provider's main feature, which is to run a single test over different inputs. E.g. a more idiomatic way to write this last test would be:
```
    @DataProvider(name = "testUmiUtilDataProvider")
    private Object[][] testUmiUtilDataProvider() {
        return new Object[][]{
                // Observed UMI / Sanitized UMI
                {"AAAA", "AAAA"},
                {"AA-AA", "AAAA"},
                {"-A-T-A", "ATA"},
                {"AAAAA--", "AAAAA"},
                {"---A", "A"},
                {"---", ""},
                {"", ""}
        };
    }

    @Test(dataProvider = "testUmiUtilDataProvider")
    public void testUmiUtil(String observed, String expected) {
        SAMRecord rec = new SAMRecord(new SAMFileHeader());
        rec.setAttribute("RX", observed);
        Assert.assertEquals(UmiUtil.getSanitizedUMI(rec, "RX"), expected);
    }
```